File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.py
class Solution:
    def trailingZeroes(self, n: int) -> int:
        if n < 5: return 0

        two, five = 0, 0

        for i in range(2, n + 1):
            if not i % 2 or not i % 5:
                chk = i
                while not (chk % 2 and chk % 5):
                    if not chk % 2:
                        chk //= 2
                        two += 1
                    if not chk % 5:
                        chk //= 5
                        five += 1

        return min(two, five)

File: 0172-factorial-trailing-zeroes/0172-factorial-trailing-zeroes.java
class Solution {
    public int trailingZeroes(int n) {
        int result = 0;

        while (n > 0) {
            if (n < 5) {
                break;
            }
            result += n / 5;
            n /= 5;
        }

        return result;
    }
}